aten/src/ATen/native/cuda/MultiLabelMarginCriterion.cu

```diff
@@ -0,0 +1,436 @@
+#include <ATen/ATen.h>
+#include <ATen/AccumulateType.h>
+#include <ATen/Dispatch.h>
+#include <ATen/NativeFunctions.h>
+#include <c10/macros/Macros.h>
+#include <ATen/cuda/CUDAContext.h>
+#include <ATen/native/cuda/block_reduce.cuh>
+
+
+namespace at {
+namespace native {
+
+namespace {
+const int MULTILABELMARGIN_THREADS = 1024;
```

Comment:
great, can you check if reducing number of threads here brings good perf and correct resulst, and we can land?